src/main/java/com/biku89/medical_clinic/PatientService.java

```diff
@@ -0,0 +1,66 @@
+package com.biku89.medical_clinic;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class PatientService {
+    private final PatientRepository patientRepository;
+    private final PatientMapper patientMapper;
+
+    public List<PatientDTO> getPatients() {
+        return patientRepository.findAll().stream().map(patientMapper::toDTO).toList();
+    }
+
+    public Optional<PatientDTO> getPatientByEmail(String email) {
+        return patientRepository.findByEmail(email).map(patientMapper::toDTO);
+    }
+
+    public PatientDTO addPatient(PatientDTO patientDTO) {
+        if (patientRepository.findByEmail(patientDTO.email()).isPresent()) {
+            throw new EmailExistingException("Patient with this email already exists");
+        }
+        Patient patient = patientMapper.toEntity(patientDTO);
+        patientRepository.save(patient);
+        return patientMapper.toDTO(patient);
+    }
+
+    public PatientDTO updatePatient(String email, PatientUpdateCommand updatedPatientDTO) {
+        Patient updatedPatient = patientMapper.toEntity(updatedPatientDTO);
+        Patient existingPatient = patientRepository.findByEmail(email)
+                .orElseThrow(() -> new PatientNotFoundException("Patient not found"));
+
+        PatientValidator.modifyingIdCardNoNotAllowed(existingPatient, updatedPatient);
+        PatientValidator.notAllowedChangeToNull(updatedPatient);
+
+        Optional<Patient> patientWithSameEmail = patientRepository.findByEmail(updatedPatient.getEmail());
+        PatientValidator.patientWithEmailIsAlreadyExist(patientWithSameEmail, updatedPatient, email);
+
+        existingPatient.setFirstName(updatedPatientDTO.getFirstName());
```

Comment:
Fajnie byłoby wynieść taki update do oddzielnej metody.
Możnaby w klasie Patient dodać metodę update:
void update(Patient patient) {
this.email = patient.getEmail();
this.birthday = patient.getBirthday();
///
}